File: servidor/fuente/RedDiff.Aplicacion/Seguridad/Usuarios/ModelosUsuarios.cs
namespace RedDiff.Aplicacion.Seguridad.Usuarios;

public sealed record CrearUsuarioSolicitud(
    string NombreUsuario,
    long RolId,
    string Contrasena);

public sealed record ActualizarUsuarioSolicitud(
    string NombreUsuario,
    long RolId,
    string? NuevaContrasena);

public sealed record CambiarEstadoUsuarioSolicitud(bool Activo);

public sealed record UsuarioResumen(
    long Id,
    string NombreUsuario,
    long RolId,
    string Rol,
    bool Activo);

public sealed record RolResumen(long Id, string Nombre, bool Activo);

File: cliente/src/app/modulos/usuarios/modelos-usuarios.ts
export interface UsuarioResumen {
  id: number;
  nombreUsuario: string;
  rolId: number;
  rol: string;
  activo: boolean;
}

export interface RolResumen {
  id: number;
  nombre: string;
  activo: boolean;
}

export interface CrearUsuarioSolicitud {
  nombreUsuario: string;
  rolId: number;
  contrasena: string;
}

export interface ActualizarUsuarioSolicitud {
  nombreUsuario: string;
  rolId: number;
  nuevaContrasena: string | null;
}
